### PR TITLE
enhancement #744: Automated index calculation

### LINKED ISSUE #744: Automate index calculation
As you can see, date and index calculations are hardcoded. This is not correct, because it does not consider all cases

image

So you should, using the logic from schedule service, automate these calculations depending on start and end dates of the semester (- 2 weeks)

### PR DESCRIPTION
**Pull request by kropli**
- added updateEventInfo method
- updated parsePair method

closes #744

### CODE REVIEW

File: src/v2/utils/parser/ScheduleParser.ts [hunk 1]
```diff
@@ -203,28 +204,55 @@ export class ScheduleParser implements Parser {
...
+        event.period === Period.EVERY_FORTNIGHT
     ) {
       await this.eventRepository.updateById(event.id, {
-        period: Period.EVERY_WEEK,
-        endTime: endOfEvent,
         eventInfo: {
           createMany: {
-            data: this.getIndexesForEventInfo(9, 17),
+            data: this.getIndexesForEventInfo((await this.getWeeksCount(
+              endOfEvent, Period.EVERY_FORTNIGHT)) + 1,
```
Comment (SharpDevOps10) [inline src/v2/utils/parser/ScheduleParser.ts:237]: Here you can use `weeksCountFortnight`

File: src/v2/utils/parser/ScheduleParser.ts [hunk 2]
```diff
@@ -203,28 +204,55 @@ export class ScheduleParser implements Parser {
...
+    await this.updateEventInfo(event, startOfEvent, endOfEvent);
+  }
+
+  async updateEventInfo (event: DbEvent, startOfEvent: Date, endOfEvent: Date) {
+    if (!event) {
+      await this.eventRepository.updateById(event.id, {
         eventInfo: {
           createMany: {
-            data: this.getIndexesForEventInfo(0, 8),
+            data: this.getIndexesForEventInfo(0, await this.getWeeksCount(endOfEvent, Period.EVERY_FORTNIGHT)),
```
Comment (SharpDevOps10) [inline src/v2/utils/parser/ScheduleParser.ts:225]: You had better create a variable `weeksCountFortnight`, because you used **getWeeksCount** method twice

**Review by SharpDevOps10 — CHANGES_REQUESTED** [review]

**Review by SharpDevOps10 — APPROVED** [review]

**Review by PavloVlasiuk — APPROVED** [review]

File: src/v2/utils/parser/ScheduleParser.ts [hunk 3]
```diff
@@ -203,28 +202,54 @@ export class ScheduleParser implements Parser {
...
+      });
+    }
+    await this.updateEventInfo(event, startOfEvent, endOfEvent);
+  }
+
+  async updateEventInfo (event: DbEvent, startOfEvent: Date, endOfEvent: Date) {
+    const weeksCountFortnight = await this.getWeeksCount(endOfEvent, Period.EVERY_FORTNIGHT);
+
+    if (!event) {
+      await this.eventRepository.updateById(event.id, {
```
Comment (fokaaas) [inline src/v2/utils/parser/ScheduleParser.ts:222]: It is not logical, you check if event exists, but take it Id

File: src/v2/utils/parser/ScheduleParser.ts [hunk 4]
```diff
@@ -203,28 +202,54 @@ export class ScheduleParser implements Parser {
...
-      event.period === Period.EVERY_FORTNIGHT
+        event.period === Period.EVERY_FORTNIGHT
     ) {
       await this.eventRepository.updateById(event.id, {
-        period: Period.EVERY_WEEK,
-        endTime: endOfEvent,
         eventInfo: {
           createMany: {
-            data: this.getIndexesForEventInfo(9, 17),
+            data: this.getIndexesForEventInfo(weeksCountFortnight + 1, await this.getWeeksCount(endOfEvent, Period.EVERY_WEEK)),
```
Comment (fokaaas) [inline src/v2/utils/parser/ScheduleParser.ts:236]: declare variable to call function

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by SharpDevOps10 — APPROVED** [review]

**Review by PavloVlasiuk — APPROVED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**